src/workers/parser/mapAwaitExpression.js

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+// @flow
+
+import { parse } from "./utils/ast";
+import generate from "@babel/generator";
+import * as t from "@babel/types";
+
+function hasTopLevelAwait(expression: string) {
+  const ast = parse(expression, { allowAwaitOutsideFunction: true });
+  const hasAwait = ast.program.body.some(
+    node =>
+      (t.isExpressionStatement(node) && t.isAwaitExpression(node.expression)) ||
+      (t.isVariableDeclaration(node) &&
+        node.declarations.some(dec => t.isAwaitExpression(dec.init)))
+  );
+
+  return hasAwait && ast;
+}
+
+function wrapExpression(ast) {
+  const statements = ast.program.body;
+  const lastStatement = statements[statements.length - 1];
+  const body = statements
+    .slice(0, -1)
+    .concat(t.returnStatement(lastStatement.expression));
+
+  const newAst = t.arrowFunctionExpression([], t.blockStatement(body), true);
+
+  return generate(newAst).code;
+}
+
+export default function handleTopLevelAwait(expression: string) {
+  const ast = hasTopLevelAwait(expression);
+  if (ast) {
+    const func = wrapExpression(ast);
+    return `(${func})().then(r => console.log(r))`;
```

Comment:
Should this also `console.error` on error?
Also, do we actually want to log the promise, and then log the result afterward, or should this include a `, void 0` at the end so that it actually evaluates to `undefined` at first or something?

Reply:
It's definitely a case of balancing between what devtools wants and what the engine actually provides :(